Libraries/Connect/Implementation/ProtocolClientConfig.swift

```diff
@@ -14,51 +14,69 @@
 
 import Foundation
 
-/// Set of configuration (usually modified through `ClientOption` types) used to set up clients.
+/// Configuration used to set up `ProtocolClientInterface` implementations.
 public struct ProtocolClientConfig {
-    /// The target host (e.g., https://buf.build).
+    /// Target host (e.g., `https://buf.build`).
     public let host: String
-    /// The client to use for performing requests.
-    public let httpClient: HTTPClientInterface
-    /// The minimum number of bytes that a request message should be for compression to be used.
-    public let compressionMinBytes: Int?
-    /// The compression type that should be used (e.g., "gzip").
-    /// Requires a matching `compressionPools` entry.
-    public let compressionName: String?
-    /// Compression pools that provide support for the provided `compressionName`, as well as any
-    /// other compression methods that need to be supported for inbound responses.
-    public let compressionPools: [String: CompressionPool]
-    /// Codec to use for serializing/deserializing requests/responses.
+    /// Protocol to use for requests and streams.
+    public let networkProtocol: NetworkProtocol
```

Comment:
I can make this rename as well

Reply:
Cool. I couldn't use protocol since it's a reserved word in swift when lowercased and it'd make ergonomics worse by having to use code tags